UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/ContractUtil.java

```diff
@@ -121,41 +135,40 @@ public GenericError getParamNumberError() {
                 .title("The given number of parameters does not match the required number of parameters for the specified transaction");
     }
 
+    public void checkMayParticipate(Context ctx, OperationData operationData) throws MissingTransactionError, LedgerAccessError, ParticipationError {
+        if (!operationData.getState().equals(OperationDataState.PENDING)) {
+            throw new ParticipationError(GsonWrapper.toJson(getOperationNotPendingError()));
+        }
+        if(!hasRightToParticipate(ctx, operationData)) {
+            throw new ParticipationError(GsonWrapper.toJson(getAccessDeniedError()));
+        }
```

Comment:
```suggestion
        if(!hasRightToParticipate(ctx, operationData)) {
            throw new ParticipationError(GsonWrapper.toJson(getAccessDeniedError()));
        }
        if (!operationData.getState().equals(OperationDataState.PENDING)) {
            throw new ParticipationError(GsonWrapper.toJson(getOperationNotPendingError()));
        }
```

rearrange so that a user that has no relation with the operation can not obtain any info about its pending state